python/GafferUI/ColorChooser.py

```diff
@@ -257,15 +257,12 @@ def __setColorInternal( self, color, reason, hsv = False ) :
 			GafferUI.NumericWidget.ValueChangedReason.DragEnd,
 		)
 
-		previousColor = self.__colorHSV if hsv else self.__color
+		colorChanged = color != ( self.__colorHSV if hsv else self.__color )
 
-		if color != previousColor or dragBeginOrEnd :
-			# we never optimise away drag begin or end, because it's important
-			# that they emit in pairs.
+		if colorChanged :
```

Comment:
This no longer updates the internal color values when `dragBeginOrEnd` is true. I don't see why that would be a problem, it seems like dragging always will result in a color change, but maybe in some odd edge cases we could be emitting the signal at L284 without an actual change?

Reply:
> maybe in some odd edge cases we could be emitting the signal at L284 without an actual change?

Yes, that is intentional. This is from memory, but I think that when dragging, the "drag end" call will always be one without an actual change, being at the same value as the previous "drag move". It's OK to skip updating `self.__color` for that (because there is no change to apply), but we don't want to skip signal emission (because clients have a reasonable expectation that there will be a drag end signalled for every drag begin).